component-tools/src/main/java/org/talend/sdk/component/tools/validator/CheckpointValidator.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (C) 2006-2025 Talend Inc. - www.talend.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.talend.sdk.component.tools.validator;
+
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.toMap;
+import static java.util.stream.Stream.empty;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import org.apache.xbean.finder.AnnotationFinder;
+import org.talend.sdk.component.api.input.Emitter;
+import org.talend.sdk.component.api.input.PartitionMapper;
+import org.talend.sdk.component.api.input.checkpoint.CheckpointAvailable;
+import org.talend.sdk.component.api.input.checkpoint.CheckpointData;
+import org.talend.sdk.component.runtime.manager.ParameterMeta;
+
+public class CheckpointValidator implements Validator {
+
+    private final Validators.ValidatorHelper helper;
+
+    public CheckpointValidator(final Validators.ValidatorHelper helper) {
+        this.helper = helper;
+    }
+
+    @Override
+    public Stream<String> validate(final AnnotationFinder finder, final List<Class<?>> components) {
```

Comment:
Should @CheckpointData  methods return type be of a @checkpont annotated class ?